modules/managing-memcached-operator-using-olm.adoc

```diff
@@ -46,26 +50,48 @@ for more information on manually defining a manifest file.
 
 . *Deploy the Operator.*
 
-.. Deploy an Operator by applying the Operator’s manifest to the desired namespace
-in the cluster:
+.. Create an OperatorGroup that specifies the namespaces that the Operator will
+target. Create the following OperatorGroup in the namespace where you will
+create the CSV. In this example, the `default` namespace is used:
++
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: memcached-operator-group
+  namespace: default
+  spec:
+    targetNamespaces:
+    - default
+----
+
+.. Apply the Operator's CSV manifest to the specified namespace in the cluster:
 +
 ----
 $ curl -Lo memcachedoperator.0.0.1.csv.yaml https://raw.githubusercontent.com/operator-framework/getting-started/master/memcachedoperator.0.0.1.csv.yaml
 $ oc apply -f memcachedoperator.0.0.1.csv.yaml
 $ oc get ClusterServiceVersion memcachedoperator.v0.0.1 -o json | jq '.status'
 ----
++
+When you apply this manifest, the cluster does not immediately update because
+it does not yet meet the requirements specified in the manifest.
 
-.. After applying this manifest, nothing has happened yet, because the cluster does
-not meet the requirements specified in our manifest. Create the RBAC rules and
-`CustomResourceDefinition` for the Memcached type managed by the Operator:
+.. Create the role, role binding, and service account to grant resource
+permissions to the Operator, and the Custom Resource Definition (CRD) to create
+the Memcached type that the Operator manages:
 +
 ----
-$ oc apply -f deploy/rbac.yaml
-$ oc apply -f deploy/crd.yaml
+$ oc create -f deploy/crds/cache_v1alpha1_memcached_crd.yaml
+$ oc create -f deploy/service_account.yaml
+$ oc create -f deploy/role.yaml
+$ oc create -f deploy/role_binding.yaml
```

Comment:
Actually, the csv object `memcachedoperator.v0.0.1` still in `Pending` status after created the above objects. The `Permissions` are inappropriate. I submitted a [PR](https://github.com/operator-framework/getting-started/pull/38) to fix it. @hasbro17 Could you help have a review? Thanks!
```console
mac:memcached-operator jianzhang$ oc describe csv memcachedoperator.v0.0.1
Name:         memcachedoperator.v0.0.1
Namespace:    default
Labels:       olm.api.7d4be2dfad49a2af=provided
Annotations:  kubectl.kubernetes.io/last-applied-configuration:
                {"apiVersion":"operators.coreos.com/v1alpha1","kind":"ClusterServiceVersion","metadata":{"annotations":{},"name":"memcachedoperator.v0.0.1...
              olm.operatorGroup: memcached-operator-group
              olm.operatorNamespace: default
              olm.targetNamespaces: default
API Version:  operators.coreos.com/v1alpha1
Kind:         ClusterServiceVersion
Metadata:
  Creation Timestamp:  2019-05-13T05:38:46Z
  Generation:          2
  Resource Version:    59654
  Self Link:           /apis/operators.coreos.com/v1alpha1/namespaces/default/clusterserviceversions/memcachedoperator.v0.0.1
  UID:                 60a17d05-7541-11e9-bebe-0a5645815f04
Spec:
  Apiservicedefinitions:
  Customresourcedefinitions:
    Owned:
      Description:   Represents a cluster of Memcached apps
      Display Name:  Memcached App
      Kind:          Memcached
      Name:          memcacheds.cache.example.com
      Resources:
        Kind:     Deployment
        Name:     
        Version:  v1beta2
        Kind:     ReplicaSet
        Name:     
        Version:  v1beta2
        Kind:     Pod
        Name:     
        Version:  v1
      Spec Descriptors:
        Description:   The desired number of member Pods for the deployment.
        Display Name:  Size
        Path:          size
        X - Descriptors:
          urn:alm:descriptor:com.tectonic.ui:podCount
      Status Descriptors:
        Description:   The current status of the application.
        Display Name:  Status
        Path:          phase
        X - Descriptors:
          urn:alm:descriptor:io.kubernetes.phase
        Description:   Explanation for the current status of the application.
        Display Name:  Status Details
        Path:          reason
        X - Descriptors:
          urn:alm:descriptor:io.kubernetes.phase:reason
      Version:   v1alpha1
  Description:   Main enterprise application providing business critical features with high availabilty and no manual intervention.
  Display Name:  Memcached Application
  Install:
    Spec:
      Deployments:
        Name:  memcached-app-operator
        Spec:
          Replicas:  1
          Selector:
            Match Labels:
              App:  memcached-app-operator
          Template:
            Metadata:
              Labels:
                App:  memcached-app-operator
            Spec:
              Containers:
                Command:
                  /usr/local/bin/memcached-operator
                Image:              quay.io/jzelinskie/memcached-operator:v0.0.1
                Image Pull Policy:  Always
                Name:               sao
                Ports:
                  Container Port:                8080
                  Protocol:                      TCP
              Restart Policy:                    Always
              Service Account:                   memcached-operator
              Service Account Name:              memcached-operator
              Termination Grace Period Seconds:  5
      Permissions:
        Rules:
          API Groups:
            *
          Resources:
            *
          Verbs:
            *
        Service Account Name:  memcached-operator
    Strategy:                  deployment
  Install Modes:
    Supported:  true
    Type:       OwnNamespace
    Supported:  true
    Type:       SingleNamespace
    Supported:  false
    Type:       MultiNamespace
    Supported:  false
    Type:       AllNamespaces
  Keywords:
    memcached
    app
  Labels:
    Alm - Owner - Enterprise - App:  memcached-app-operator
    Alm - Status - Descriptors:      memcached-app-operator.v0.0.1
  Maintainers:
    Email:   corp@example.com
    Name:    Some Corp
  Maturity:  alpha
  Provider:
    Name:   Example
    URL:    www.example.com
  Version:  0.0.1
Status:
  Certs Last Updated:  <nil>
  Certs Rotate At:     <nil>
  Conditions:
    Last Transition Time:  2019-05-13T05:38:46Z
    Last Update Time:      2019-05-13T05:38:46Z
    Message:               requirements not yet checked
    Phase:                 Pending
    Reason:                RequirementsUnknown
    Last Transition Time:  2019-05-13T05:38:46Z
    Last Update Time:      2019-05-13T05:38:46Z
    Message:               one or more requirements couldn't be found
    Phase:                 Pending
    Reason:                RequirementsNotMet
  Last Transition Time:    2019-05-13T05:38:46Z
  Last Update Time:        2019-05-13T05:38:46Z
  Message:                 one or more requirements couldn't be found
  Phase:                   Pending
  Reason:                  RequirementsNotMet
  Requirement Status:
    Group:    operators.coreos.com
    Kind:     ClusterServiceVersion
    Message:  CSV missing minimum kube version specification
    Name:     memcachedoperator.v0.0.1
    Status:   NotPresent
    Version:  v1alpha1
    Group:    apiextensions.k8s.io
    Kind:     CustomResourceDefinition
    Message:  CRD is present and Established condition is true
    Name:     memcacheds.cache.example.com
    Status:   Present
    Uuid:     c6c40b90-7540-11e9-bebe-0a5645815f04
    Version:  v1beta1
    Dependents:
      Group:    rbac.authorization.k8s.io
      Kind:     PolicyRule
      Message:  namespaced rule:{"verbs":["*"],"apiGroups":["*"],"resources":["*"]}
      Status:   NotSatisfied
      Version:  v1beta1
    Group:      
    Kind:       ServiceAccount
    Message:    Policy rule not satisfied for service account
    Name:       memcached-operator
    Status:     PresentNotSatisfied
    Version:    v1
Events:
  Type    Reason               Age                From                        Message
  ----    ------               ----               ----                        -------
  Normal  RequirementsUnknown  13m                operator-lifecycle-manager  requirements not yet checked
  Normal  RequirementsNotMet   13m (x2 over 13m)  operator-lifecycle-manager  one or more requirements couldn't be found
```

Reply:
@jianzhangbjz Yes, the following rule was incorrect for a Role:
```YAML
- apiGroups:
          - "*"
...
```
I've merged your PR. Were you able to verify the CSV with that fix?
If not then I can go through it again.

Reply:
Sure, thanks!